examples/src/main/scala/com/cloudera/livy/apps/WordCountApp.scala

```diff
@@ -0,0 +1,154 @@
+      throw new FileNotFoundException("Examples App jar not found in the target folder")
+    }
+    val scalaApiPath = scalaClient.getClass.getProtectionDomain.getCodeSource.getLocation.getPath
+    val scalaApiJarUploadFuture = scalaClient.uploadJar(new File(scalaApiPath))
+    scalaApiJarUploadFuture onComplete {
+      case Success(t) => println("Successfully uploaded jar scala-api")
+      case Failure(e) => throw e
+    }
+    val appJarUploadFuture = scalaClient.uploadJar(new File(appJarPath))
+    appJarUploadFuture onComplete {
+      case Success(t) => println("Successfully uploaded jar examples::")
+      case Failure(e) => throw e
+    }
+  }
+
+  def processStreamingWordCount(host: String, port: Int, outputPath:String): Unit = {
+    val handle = scalaClient.submit { context =>
+      context.createStreamingContext(20000)
+      var index = 1
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyConent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      if (index == 2) {
+        ssc.stop(true, true)
+      } else {
+        index += 1
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(40000)
+    }
+  }
+
+  def getWordWithMostCount(outputPath: String): ScalaJobHandle[Any] = {
+    val handle = scalaClient.submit { context =>
+      val sqlctx = context.sqlctx
+      val rdd = sqlctx.read.json(outputPath)
+      rdd.registerTempTable("words")
+      val result = sqlctx.sql("select word, count(word) as word_count from words " +
+        "group by word order by word_count desc limit 1")
+      if (result != null && result.count() > 0) {
+        result.collect()(0)
+      }
+    }
+    handle
+  }
+
+  private def filterEmptyConent(text: String): Boolean = {
+    if (text == null || text.isEmpty) false else true
+  }
+
+  private def tokenize(text : String) : Array[String] = {
+    text.toLowerCase.replaceAll("[^a-zA-Z0-9\\s]", "").split("\\s+")
+  }
+
+  /**
+    * Main method of the WordCount App
+    * STEPS FOR EXECUTION
+    * 1) 'livy.file.local-dir-whitelist' config has to be set in livy.conf file for the jars to be
+    * added to the session
+    *
+    * 2) Libraries to be added to classpath - livy-api-version-SNAPSHOT.jar,
+    * livy-scala-api-version-SNAPSHOT.jar, livy-client-http-version-SNAPSHOT.jar
+    *
+    * @param args
+    * arg(0) - Livy server url
+    * arg(1) - Output path to save the text read from the stream
+    */
+  def main(args: Array[String]) {
+    if (args.length < 2) {
+      throw new IllegalArgumentException("Number of args is less")
+    }
+    val outputPath = args(1)
+    val lock = new Object
+    lock.synchronized {
+      init(args(0))
+      lock.wait(12000)
```

Comment:
What are these waits accomplishiog?

Remember what I mentioned before: every time you do a wait without a notify, or you call `Thread.sleep`, you're most probably doing something you shouldn't, and instead you should actually be waiting for something to happen.